app/jobs/hyku_addons/reindex_model_job.rb

```diff
@@ -51,9 +46,37 @@ def reindex_works(works)
       def mint_doi(work)
         return if work.doi.present?
 
+        Rails.logger.debug "=== about to mint doi for #{work.title} ==== "
         work.update(doi_status_when_public: "findable")
         register_doi = Hyrax::DOI::DataCiteRegistrar.new.register!(object: work)
         work.update(doi: [register_doi.identifier])
       end
+
+      def fetch_work_using_ids(klass)
+        @use_work_ids.map do |id|
+          work = klass.find(id)
+          Rails.logger.debug "=== updating index with #{id}for #{work&.title&.to_a&.first} ===="
+          mint_doi(work) if @cname_doi_mint.present? && @cname_doi_mint.include?(@cname)
+          work.save
+        end
+      end
+
+      # When the offset becomes too large, no records would be found
+      def reindex_and_mint_work(work_class, offset)
+        works = work_class.where("title_tesim:*").limit(@limit).offset(offset)
+        return unless works&.to_a&.any?
+
+        Rails.logger.debug "=== Starting to reindex #{@klass} in #{@cname} ==="
+
+        # with calling to_a it always returns true, even when no records found
+        reindex_works(works)
+
+        new_page_count = @page.to_i + 1
+
+        # Re-enqueue
+        ReindexModelJob.perform_later(@klass, @cname, @limit, page: new_page_count, cname_doi_mint: @cname_doi_mint)
+        sleep(0.5)
```

Comment:
I'm not sure we should be adding sleep's to the jobs, this seems like a smell to me.

Reply:
I am happy to remove but what makes it a smell

Reply:
Especially not after asking a job to `perform_later`.

Reply:
Just adding in sleep within a background job. A job should ideally just perform a single task, and if there are more tasks, each should be delegated. So having a sleep here, would imply that the job is doing a lot of work that needs to be broken up to prevent the site's performance from being hindered.

Reply:
I agree.  You enqueue a job to be processed later, then wait half a second, then finish writing to the log.  What /does/ happen in that half a second, and why is it so important to sleep before writing your final comment to the log? There should be no reason to wait to write to the log, and it indicates a fragility or failure elsewhere.